grisp/default/common/build/nifs/grisp_rtems_nif.c

```diff
@@ -116,12 +118,52 @@ static ERL_NIF_TERM unmount_nif(ErlNifEnv *env, int argc,
   return enif_make_atom(env, "ok");
 }
 
+static ERL_NIF_TERM pwrite_nif(ErlNifEnv *env, int argc,
+                               const ERL_NIF_TERM argv[]) {
+  int fd;
+  int size, offset;
+  int bytes_written;
+  int close_status;
+  ErlNifBinary device_path, buffer;
+
+  if (!enif_inspect_iolist_as_binary(env, argv[0], &device_path))
+    return enif_make_badarg(env);
+
+  if (!enif_inspect_iolist_as_binary(env, argv[1], &buffer))
+    return enif_make_badarg(env);
+
+  if (!enif_get_int(env, argv[2], &offset))
+    return enif_make_badarg(env);
+
+  fd = open(device_path.data, O_RDWR);
+  if (fd < 0)
+    return enif_make_tuple3(
+        env, enif_make_atom(env, "error"), enif_make_atom(env, "open"),
+        enif_make_string(env, strerror(errno), ERL_NIF_LATIN1));
```

Comment:
I would save the value of `errno` into some variable before any other function call. I don't know which function will be called first here and whether it might change `errno`. I think in general it's good practice to directly save it, when one wants to use the value.

Reply:
Just to add to this: call order is unspecified, hence this makes sense.